x-pack/filebeat/input/internal/private/private.go

```diff
@@ -35,7 +35,11 @@ var privateKey = reflect.ValueOf("private")
 // `private:""`, the fields with the tag will be marked as private. Otherwise
 // the comma-separated list of names with be used. The list may refer to its
 // own field.
-func Redact[T any](val T, tag string, global []string) (redacted T, err error) {
+func Redact[T any](val T, tag string, global []string, replace ...Replacer) (redacted T, err error) {
```

Comment:
In the names of these parameters, variables, functions and types:

```
replace
Replacer
rep
compileReplacers
map[reflect.Type]func(reflect.Value) reflect.Value
replacers
[]Replacer
reps
```

it would be good to distinguish more clearly between:
- singular vs plural
- a `Replacer` vs a `Replacer` wrapped to handle `reflect.Value`
- a slice of `Replacer` vs a map of `reflect.Type` to `Replacer` wrapped to handle `reflect.Value`

Maybe names like `rawReplacers` / `wrappedReplacers`, and `replacersByType` / `wrappedReplacersByType` / `compiledRepalcers` would help. Or maybe it could be a bit less explicit than that, but I think tweaking the names here would really make it easier to follow.

Reply:
There is a system here. Note that labels can be verbs as well and so may not have (suffix-marked) number.

In the case of the `Redact` call, `replace` is a verb, and in `compileReplacers` is also, in order to match since it's the only place where it's used.

I've fixed the number for the return from `compileReplacers`, that was a mistake. I've also made the label match within the body of that function and used the named type (the latter also a mistake).

> a `Replacer` vs a `Replacer` wrapped to handle reflect.Value

I've improved the locality to make this clearer; all the relevant code now fits in a single screen.

I don't want to use add globally contextual naming to things that can be clear from context; when you see `rawReplacer` it immediately makes you need to now what the `raw` refers to, which may not be helpful. The type annotation IMO is enough to indicate intent.

Reply:
I like the changes.

> I don't want to use add globally contextual naming to things that can be clear from local context

:+1: Agreed.

> when you see `rawReplacer` it immediately makes you need to now what the `raw` refers to

Good point.